installation: Update password check for live CD issue, old soft failure, remove sleep

okurz opened the pull request.

okurz commented:
@asmorodskyi

asmorodskyi commented on tests/installation/user_settings.pm:
```diff
@@ -43,13 +41,17 @@ sub run() {
+            record_soft_failure 'boo#1013206';
+        }
+        else {
+            record_soft_failure 'bsc#937012';
```
We need to die here , not soft fail on closed bug

coolo replied:
In maintenance testing we also install old products. After you checked they are not affected, I'm fine with a die

coolo replied:
bingo -> https://openqa.suse.de/tests/648953#step/user_settings/6

coolo replied:
so skip this on check_var('VERSION', '12') && check_var('ARCH', 's390x') and assert everywhere else.

okurz replied:
done

okurz commented:
wait, I need to check the user_settings_root

okurz commented:
Updated:
* Extracted common methods for `user_settings` and `user_settings_root` with a new approach, multiple inheritance to include a base class in both cases

Local verification run: http://lord.arch/tests/5432
